src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs

```diff
@@ -33,18 +38,27 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
             //
             // The imports come logically before the main razor file and are in the order they
             // should be processed.
-            var visitor = new DirectiveVisitor(descriptors);
+            DirectiveVisitor visitor = null;
+            if (FileKinds.IsComponent(codeDocument.GetFileKind()))
+            {
+                var currentNamespace = ComputeNamespace(codeDocument);
```

Comment:
I'm OK-ish with this being here for now. It just reinforces things we already know should be improved. The namespace of the generated code isn't supposed to be a parser concern.